RUFAS/output_manager.py

```diff
@@ -301,18 +301,108 @@ def _generate_file_name(self, base_name: str, extension: str = "json") -> str:
         return f"{base_name}_{timestamp}.{extension}"
 
+    def _exclude_info_maps(self, pool: Dict[str, pool_element_type]) -> Dict[str, pool_element_type]:
+        """ Makes a copy of the given pool and removes info_maps from it.
+
+        Returns
+        -------
+        Dict[str, OutputManager.pool_element_type]
+            A copy of the given pool with info_maps removed from it.
+
+        """
+
+        pool_copy = pool.copy()
+        for key, value in pool_copy.items():
+            if isinstance(value, dict) and "info_maps" in value:
+                value.pop("info_maps")
+        return pool_copy
+
+    def _load_txt_file_to_list(self, path: str) -> List[str]:
+        """ Reads a text file into a list.
+
+        Parameters
+        ----------
+        path : str
+            Path of the input file to be read.
+
+        Returns
+        -------
+        List
+            A list of keys the user has selected to filter the variables pool.
+
+        Raises
+        -------
+        Exception
+            If an error occurs while opening or reading the file.
+
+        """
+        try:
+            with open(path) as keys_doc:
+                return keys_doc.read().splitlines()
+        except Exception as e:
+            raise e
+
+    def _filter_variables_pool(self, inclusion_keys: List[str]):
+        """
+        Takes the list of keys the user wants in their final data pool,
+        filters the variables pool accordingly, and returns the filtered pool.
+
+        Parameters
+        ----------
+        inclusion_keys : List[str]
+            A list of keys the user has selected to filter the variables pool.
+
+        Returns
+        -------
+        Dict[str, OutputManager.pool_element_type]
+            The variables_pool with only the values paired with the keys
+            from the inclusion_keys list remaining.
+
+        """
+        return {key: self.variables_pool[key] for key in inclusion_keys if key in self.variables_pool.keys()}
+
+    def save_variables(self, path: str, keys_file_path: str,
+                       exclude_info_maps: bool = False) -> None:
+        """
+        Reads a text file containing a list of keys and filters the variables pool by those keys.
+        Saves resulting data pool to a json file in the given path to a directory.
+
+        Parameters
+        ----------
+        path : str
+            Path to the directory where the file will be saved.
+
+        keys_file_path : str
+            Path of the input file containing the list of keys.
+
+        """
+        inclusion_keys = self._load_txt_file_to_list(keys_file_path)
+        filtered_pool = self._filter_variables_pool(inclusion_keys)
+        final_pool = filtered_pool
+        if exclude_info_maps:
+            final_pool = self._exclude_info_maps(filtered_pool)
+        file_path = os.path.join(path, self._generate_file_name("saved_variables", "json"))
+        self._dict_to_file_json(final_pool, file_path)
```

Comment:
I know I wrote this code, but now that I'm looking at it, the extra variable is not needed.
```suggestion
        if exclude_info_maps:
            filtered_pool = self._exclude_info_maps(filtered_pool)
        file_path = os.path.join(path, self._generate_file_name("saved_variables", "json"))
        self._dict_to_file_json(filtered_pool, file_path)
```